src/driver/mysql/MysqlDriver.ts

```diff
@@ -640,12 +640,13 @@ export class MysqlDriver implements Driver {
             // console.log("isUnique:", tableColumn.isUnique, this.normalizeIsUnique(columnMetadata));
             // console.log("isGenerated:", tableColumn.isGenerated, columnMetadata.isGenerated);
             // console.log("==========================================");
-
             return tableColumn.name !== columnMetadata.databaseName
                 || tableColumn.type !== this.normalizeType(columnMetadata)
                 || tableColumn.length !== columnMetadata.length
-                || tableColumn.width !== columnMetadata.width
-                || tableColumn.precision !== columnMetadata.precision
+                || (tableColumn.width !== columnMetadata.width
+                    && (tableColumn.width || columnMetadata.width))
```

Comment:
a bit complex expression. Can you please explain it?

Reply:
We strict check if values aren't equals, but column is changed when at least one of the values is defined. For this case undefined and null should be treated as equals(didn't want to use ```==```).

I'm not sure atm which one returned ```undefined``` and which ```null```. I can check that out and find out why there are different if that's necessary.

Reply:
Solution looks a bit complex and it looks like the root of the problem is somewhere else. Maybe we should figure out why its somewhere is undefined and somewhere is null?

Reply:
I guess first few bugfixes gonna be hard :)
It looks like this is not necessary for width. Precision is more problematic - it accepts nulls(#609 cc789b410eb9a0e962101b22175284da37b5f387). 

In case of MySql driver when that check is executed `tableColumn.precision=0` and `columnMetadata.precision=null`. The question is what `tableColumn.precision` value should be - should we change it always to null when it is 0 on time columns? It is set on line https://github.com/typeorm/typeorm/blob/master/src/driver/mysql/MysqlQueryRunner.ts#L1283 

On other drivers there is comparison between `tableColumn.precision=undefined` and `columnMetadata.precision=null`(because we set it explicitly to null in entity options).

Reply:
Is it possible for precision to be equal to zero? If no then maybe we can replace null with zero. If yes, then maybe we'll use undefined values instead of nulls everywhere ? (except here in this edge-case precision case)

Reply:
mysql acts a bit strange for my taste:
- timestamp with precision = 0 - creates timestamp without precision
- decimal with precision = 0 - creates decimal with precision 10(default)

I see two options how to solve it:
- when reading database state set column precision to null if db returns 0 - it will break synchronization for time columns with precision set to 0(in entity definition), ppl would have to change precision to null(setting precision to 0 doesn't work on mysql<5.6)
- deprecate null as precision value in column options(metadata) and  when syncing db add columns without passing precision value if precision is set to 0(current implementation for null precision)

Reply:
sounds okay to me. Just make sure you know all the cases when precision can be zero or not for all databases. Schema synchronization is a fragile place, we need to be careful with changes.

Reply:
Ok, I think I will go with first option - I guess more people use precision = null than precision = 0.
Should it(logic about null precision) be implemented in all of the drivers or just mysql(where that edge case really exists)? In other words do we want `precision:null` to be accepted anywhere or just on old mysql servers where it was intended to do?

Reply:
@vlapo 
Default values are declared in `dataTypeDefaults` and mostly handled in a different place(internal structures don't contain properties with default values most of the time) so I wouldn't put them to this PR.
Of course writing more tests would be helpful, but it takes a lot of time before making actual changes to the code - I'm not sure we've got enough time since none of us works full-time on the project atm. I think fixing one problem at the time might be a better solution for someone who doesn't really know how much time he/she would be able to contribute next week/month.

Reply:
I can implement second one if anyone likes it more. I don't really like any of them :)
First one keeps nulls as precision so nobody would have to change theirs entity schemas after upgrading. But leaves a question how null should be handled on other dbs.
Second one introduce 0 as a magic number - depending on column type it will act differently. But in a long run it might be better, because we wouldn't allow null as precision for any(others) db driver.


After a little thought I guess my choice may be due to how such cases are handled at daily job - no depreciation/breaking change policy(which isn't a good policy).

Reply:
I prefer second approach. We just need to make sure if can always omit precision for drivers if its equal to zero (e.g. make sure there is no "0" as a precision value for any driver we have).

Regarding to depreciation/breaking change policy - currently we try to be flexible and at the same time avoid critical breaking changes in master. I don't think we should follow depreciation strategy so much at the moment. It all takes time, and time is limited, so I prefer to stay flexible instead.

Reply:
Checked every type with precision on mssql, mysql, postgres - either acts like no precision was specified or throws an error when trying to create a column. Sqlite however allows columns with precision = 0.

Reply:
Since Extended Support for Mysql 5.5 Ends this December do we actually need `precision: null` anymore? I'm not sure if we should support if it is eol. However it would be breaking for folks who still use 5.5 so, probably should be against next if we decide to remove that option.

http://www.oracle.com/us/support/library/lifetime-support-technology-069183.pdf Oracle’s MySQL Releases

Reply:
Let me ask people is slack. But yeah probably we need to drop it then.

Reply:
Okay lets drop 5.5 support.